### PR TITLE
Methods returning array

### PR DESCRIPTION
**Pull request by rpavlik**
Turns out this wasn't implemented yet. Wasn't too hard, even got signature deduction figured out.

### CODE REVIEW

File: tests/main.cpp [hunk 1]
```diff
@@ -304,6 +304,21 @@ TEST(Object_call_byNameWithArgs)
     ASSERT(str2.call<wchar_t>("charAt", 1) == L'e');
 }
 
+TEST(Object_call_returningArray) {
+    jni::Object str = jni::Class("java/lang/String").newInstance("Testing");
+
+    {
+        auto getBytes =
+            jni::Class("java/lang/String").getMethod("getBytes", "()[B");
+        auto bytes = str.call<jni::Array<jni::byte_t>>(getBytes);
```
Comment (mitchdowd) [inline tests/main.cpp:313]: To be working properly we'd want signature generation, so you can do:

auto bytes = `str.call<Array<byte_t>>("getBytes");`

It helps out those who don't necessarily know the JNI method sig format.

Comment (rpavlik) [reply tests/main.cpp:313]: It does have that now, just not in the first version I pushed. I thought I had unit tests for both, maybe I missed the second?

Comment (rpavlik) [reply tests/main.cpp:313]: Yeah I just missed it in the tests. You can see with the valueSig changes that I did get that working.

Comment (rpavlik) [reply tests/main.cpp:313]: The tests for the signature generation version are right below.

**Pull request merged**